peas/weather_abstract.py

```diff
@@ -0,0 +1,214 @@
+                sky_safe = True
+
+        self.logger.debug('Cloud Condition: {} (Sky-Amb={:.1f} C)'.format(cloud_condition, last_cloud))
+
+        return cloud_condition, sky_safe
+
+    def _get_wind_safety(self, wind_speed, wind_gust):
+        safety_delay = self.safety_delay
+
+        wind_thresholds = self.thresholds['wind_speed']
+
+        threshold_windy = wind_thresholds.get('windy')
+        threshold_very_windy = wind_thresholds.get('very_windy')
+
+        gust_thresholds = self.thresholds['wind_gust']
+
+        threshold_gusty = gust_thresholds.get('gusty')
+        threshold_very_gusty = gust_thresholds.get('very_gusty')
+
+        if len(wind_speed) == 0:
+            self.logger.debug('UNSAFE: no average wind speed readings found')
+            wind_condition = 'Unknown'
+            wind_safe = False
+        else:
+            # Windy?
+            if wind_speed > threshold_very_windy[0] and wind_speed <= threshold_very_windy[1]:
+                self.logger.debug('UNSAFE:  Very windy in last {:.0f} min. Average wind speed {:.1f} kph'.format(
+                                  safety_delay, wind_speed))
+                wind_condition = 'Very Windy'
+                wind_safe = False
+            elif wind_speed > threshold_windy[0] and wind_speed <= threshold_windy[1]:
+                self.logger.debug('UNSAFE:  Windy in last {:.0f} min. Average wind speed {:.1f} kph'.format(
+                                  safety_delay, wind_speed))
+                wind_condition = 'Windy'
+                wind_safe = False
+            else:
+                wind_condition = 'Calm'
+                wind_safe = True
+
+            self.logger.debug('Wind Condition: {} ({:.1f} km/h)'.format(wind_condition, wind_speed))
+
+        if len(wind_gust) == 0:
+            self.logger.debug('UNSAFE: no maximum wind gust readings found')
+            gust_condition = 'Unknown'
+            gust_safe = False
+        else:
+            # Gusty?
+            if wind_gust > threshold_very_gusty[0] and wind_gust <= threshold_very_gusty[1]:
+                self.logger.debug('UNSAFE:  Very gusty in last {:.0f} min. Max gust speed {:.1f} kph'.format(
+                                  safety_delay, wind_gust))
+                gust_condition = 'Very Gusty'
+                gust_safe = False
+            elif wind_gust > threshold_gusty[0] and wind_gust <= threshold_gusty[1]:
+                self.logger.debug('UNSAFE:  Very gusty in last {:.0f} min. Max gust speed {:.1f} kph'.format(
+                                  safety_delay, wind_gust))
+                gust_condition = 'Gusty'
+                gust_safe = False
+            else:
+                gust_condition = 'Calm'
+                gust_safe = True
+
+            self.logger.debug('Gust Condition: {} ({:.1f} km/h)'.format(gust_condition, wind_gust))
+
+        return (wind_condition, wind_safe), (gust_condition, gust_safe)
+
+    def _get_status(self):

```

Comment:
You need to actually use this method!

Reply:
For consistency with other methods should probably take `current_values` as an argument, instead of poking around in `weather_entries` itself.

Reply:
Done.